### PR TITLE
Micronaut Framework 4.0.x 

### PR DESCRIPTION
**Pull request by sdelamo**
This make 4.0.x tests pass. 

I had to ignore several tests. I will will revert in individual PRs. 

- [x]  [ignore azure error](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/8cc574af104e203d14b3fc26b434d287b53b66da)  
- [x] [ignore HttpStatusHandler test](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/cd1115dbbd93dcdfb167dd4a0abcd05bd3d8e2de)
- [x]  [ignore messaging kafka-streams and jms-oracle-aq](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/2642e27893a6b55c4aaef89b7f8eb4aec11e4c61)
- [x] [ignore community libraries](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/5d48d7c9efd4c95628aa300fa0a953b36c0be868)
- [x] [ignore maven](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/69dce54004d0b0639c15ac97cd8f6fb9613b766c)
- [x] [ignore functional AmazonCognitoSpec until security is republished](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/bf5c12d949820a2d041dcb5312312bddf158e2ba)
- [x] [ignore: aws-lambda custom runtime for groovy](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/476cc913dfe95f78cb7a0f39f9f0f6537166291d)  …
- [x]  [ignore due to findArgumentBinder](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/997b1b0f2b6205eacaa499788e25ebb94c009d43)
- [x] [ignore oracle for hibernate reactive](https://github.com/micronaut-projects/micronaut-starter/pull/1723/commits/c4dc4799e592ea94ba567e7d330226d571f1d2cf)

Moreover, the native image generation still fails. I will create a separate issue for it.

### CODE REVIEW

**Comment by timyates** [comment]
- native builds are failing as we LoggerFactory [was initialized at build time](https://github.com/micronaut-projects/micronaut-starter/actions/runs/4767330472/jobs/8475432555?pr=1723#step:5:41)
- functest-aws has [some failures](https://ge.micronaut.io/s/k2t2c3twhc5yg/tests/overview?outcome=failed)
- functest-core has [failures WRT Kapt](https://ge.micronaut.io/s/esgnxxhuoyfuq/tests/overview?outcome=failed)

😿

File: starter-core/build.gradle [hunk 1]
```diff
@@ -14,23 +14,26 @@ sourceSets {
 }
 
 dependencies {
-    api platform("io.micronaut:micronaut-bom:$micronautVersion")
+    api platform("io.micronaut.platform:micronaut-platform:$micronautVersion")
```
Comment (timyates) [inline starter-core/build.gradle:17]: Should we be pulling platform here?

Or micronaut-core-bom?

File: starter-core/src/test/groovy/io/micronaut/starter/feature/kotlin/KtorSpec.groovy [hunk 2]
```diff
@@ -73,50 +76,24 @@ class KtorSpec extends ApplicationContextSpec implements CommandOutputFixture {
...
-      <groupId>io.micronaut.kotlin</groupId>
-      <artifactId>micronaut-kotlin-runtime</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        and:
-        !template.contains('<artifactId>micronaut-http-server-netty</artifactId>')
+        verifier.hasDependency("io.micronaut.kotlin", "micronaut-ktor", Scope.COMPILE)
+        verifier.hasDependency("io.ktor", "ktor-server-netty", Scope.COMPILE)
+        verifier.hasDependency("io.ktor", "ktor-serialization-jackson", Scope.COMPILE)
```
Comment (timyates) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/kotlin/KtorSpec.groovy:91]: Gotta love people renaming modules ;-)

File: starter-web-servlet/build.gradle [hunk 3]
```diff
@@ -11,11 +11,12 @@ configurations {
 }
 
 dependencies {
-    developmentOnly platform("io.micronaut:micronaut-bom:$micronautVersion")
+    developmentOnly platform("io.micronaut.platform:micronaut-platform:$micronautVersion")
```
Comment (timyates) [inline starter-web-servlet/build.gradle:14]: again, not sure if we should be pulling the platform or just the bom 🤔

**Review by timyates — APPROVED** [review]
As far as I can tell, this is good progress...

We mustn't forget the now ignored `DiffControllerSpec.groovy` and `ZipCreateControllerSpec.groovy` tests...

And the Kapt failure is concerning...

I'm not 100% sure pulling the micronaut-platform in as a dependency is the correct choice... I guess it should be ok though, as platform does not depend on starter...  In other modules, we've made-do with pulling in `micronaut-core-bom`, but the situation here is probably different

File: starter-core/build.gradle (hunk 1 above)
Comment (sdelamo) [reply starter-core/build.gradle:17]: platform.

File: starter-core/src/test/groovy/io/micronaut/starter/feature/kotlin/KtorSpec.groovy (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/kotlin/KtorSpec.groovy:91]: I think we should make those maven coordinates managed in Kotlin.

File: starter-web-servlet/build.gradle (hunk 3 above)
Comment (sdelamo) [reply starter-web-servlet/build.gradle:14]: Platform.

**Pull request merged**